arch/x86_64/Kconfig

```diff
@@ -0,0 +1,4 @@
+gsource "arch/x86_64/soc/*/Kconfig.soc"
```

Comment:
Could use a plain `source` here now. I recently improved how globbing is handled, and `gsource` is only supported for backwards compatibility now: https://github.com/zephyrproject-rtos/zephyr/pull/9415 (the commit message has a motivation)